config/database.go

```diff
@@ -219,48 +221,78 @@ func (ds *DatabaseStore) persist(cfg *model.Config) error {
 		return errors.Wrap(err, "failed to serialize")
 	}
 
-	id := model.NewId()
 	value := string(b)
-	createAt := model.GetMillis()
-
 	err = ds.checkLength(len(value))
 	if err != nil {
 		return errors.Wrap(err, "marshalled configuration failed length check")
 	}
 
+	sum := sha256.Sum256(b)
+
+	// Skip the persist altogether if we're effectively writing the same configuration.
+	var oldValue string
+	var row *sql.Row
+	if ds.driverName == model.DatabaseDriverMysql {
+		// mysql ignores active index
```

Comment:
We can be a bit more verbose here :)

Something like:

> We use a sub-query to get the id first because selecting the id column using active uses the index, but selecting sha column using active does _not_ use the index. That's why we get the id first, and then get the sha using the id. The sub-query uses the active index, and then the top-level query uses the primary key. This takes 2 queries, but it is actually faster than one slow query for MySQL

Reply:
Ehe, added your words :p